src/test/java/org/skriptlang/skript/test/tests/lang/ConditionalTest.java

```diff
@@ -0,0 +1,368 @@
+package org.skriptlang.skript.test.tests.lang;
+
+import ch.njol.skript.lang.util.ContextlessEvent;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.skriptlang.skript.lang.condition.CompoundConditional;
+import org.skriptlang.skript.lang.condition.Conditional;
+
+public class ConditionalTest {
+
+	TestConditional condTrue;
+	TestConditional condFalse;
+	TestConditional condUnknown;
+	Event event;
+
+	@Before
+	public void setup() {
+		condTrue = new TestConditional(Kleenean.TRUE);
+		condFalse = new TestConditional(Kleenean.FALSE);
+		condUnknown = new TestConditional(Kleenean.UNKNOWN);
+		event = ContextlessEvent.get();
+	}
+
+	@Test
+	public void testBasicConditionals() {
+		Assert.assertEquals(Kleenean.TRUE, condTrue.evaluate(event));
+		Assert.assertEquals(1, condTrue.timesEvaluated);
+		condTrue.reset();
+
+		Assert.assertEquals(Kleenean.FALSE, condFalse.evaluate(event));
+		Assert.assertEquals(1, condFalse.timesEvaluated);
+		condFalse.reset();
+
+		Assert.assertEquals(Kleenean.UNKNOWN, condUnknown.evaluate(event));
+		Assert.assertEquals(1, condUnknown.timesEvaluated);
+		condUnknown.reset();
+	}
+
+	private void assertBasic(TestConditional conditional, Kleenean expected, Kleenean actual, int expectedEvals) {
+		Assert.assertEquals("Incorrect evaluation!", expected, actual);
+		assertEvals(conditional, expectedEvals);
+	}
+
+	private void assertEvals(TestConditional conditional, int expectedEvals) {
+		Assert.assertEquals("Wrong number of evaluations!", expectedEvals, conditional.timesEvaluated);
+		conditional.reset();
+	}
+
+	@Test
+	public void testBasicAndKnown() {
+		// true AND known x
+		assertBasic(condTrue, Kleenean.TRUE, condTrue.and(Kleenean.TRUE, event), 1);
+		assertBasic(condTrue, Kleenean.FALSE, condTrue.and(Kleenean.FALSE, event), 0);
+		assertBasic(condTrue, Kleenean.UNKNOWN, condTrue.and(Kleenean.UNKNOWN, event), 1);
+
+		// false AND known x
+		assertBasic(condFalse, Kleenean.FALSE, condFalse.and(Kleenean.TRUE, event), 1);
+		assertBasic(condFalse, Kleenean.FALSE, condFalse.and(Kleenean.FALSE, event), 0);
+		assertBasic(condFalse, Kleenean.FALSE, condFalse.and(Kleenean.UNKNOWN, event), 1);
+
+		// unknown AND known x
+		assertBasic(condUnknown, Kleenean.UNKNOWN, condUnknown.and(Kleenean.TRUE, event), 1);
+		assertBasic(condUnknown, Kleenean.FALSE, condUnknown.and(Kleenean.FALSE, event), 0);
+		assertBasic(condUnknown, Kleenean.UNKNOWN, condUnknown.and(Kleenean.UNKNOWN, event), 1);
+	}
+
+	@Test
+	public void testBasicOrKnown() {
+		// true AND known x
```

Comment:
am i misunderstanding this comment or should it be `OR` not `AND`

Reply:
Yeah mistake